.github/pull_request_template.md

```diff
@@ -0,0 +1,30 @@
+<!-- Provide a general summary of your changes in the Title above -->
+
+### Description
+<!--
+Describe your changes in detail.
+For contribution process guide, look into CONTRIBUTING.md
+
+If this is a relatively large or complex change:
+ - BEFORE creating a PR, find existing or start a new discussion on Issues tab,
+ - if the discussion is concluded, go ahead with this PR,
+ - perhaps describe what alternatives you considered.
+
+If this PR references or fixes an open issue, please link it here
+    using "Ref. #<number>" or "Fixes: #<number>".
+-->
+
+### Checklist
+<!--- Put an 'x' in the boxes that are checked -->
+
+- [ ] Code compiles without errors locally
+- [ ] All tests pass locally
+- [ ] CI workflows executes properly, incl. workflows not executed per PR (e.g. Nightly)
+<!-- If you have more tasks to do before merging this PR, simply add them here -->
+
+<!-- You can remove these entries, if they don't apply -->
+- [ ] New tests added, especially if they will fail without my changes
+- [ ] Added/extended example(s) to cover this functionality
+- [ ] Extended the README/documentation
+- [ ] All newly added files have a license
+- [ ] All newly added files are referenced in CMake files
```

Comment:
nit: Code is formatted in alignment with `clang-format`.

Reply:
yeah, I was thinking about mentioning code style (especially since we have such section in CONTRIBUITING.md), but it's covered by CI being green - do you think it's worth mentioning it anyway? we should then add info about all our checks...?

Reply:
Alright, it's included in `CI workflows executes properly, incl. workflows not executed per PR (e.g. Nightly)`.